lib/charms/opensearch/v0/helper_networking.py

```diff
@@ -25,9 +26,22 @@
 logger = logging.getLogger(__name__)
 
 
-def get_host_ip(charm: CharmBase, peer_relation_name: str) -> str:
+class NetworkType(str, Enum):
+    """NetworkType Enum."""
+
+    PUBLIC = "public"
+    PRIVATE = "private"
+    BIND = "bind"
+
+
+def get_host_ip(charm: CharmBase, relation_name: str, type: NetworkType = NetworkType.BIND) -> str:
     """Fetches the IP address of the current unit."""
-    address = charm.model.get_binding(peer_relation_name).network.bind_address
+    if type == NetworkType.PRIVATE:
+        address = charm.model.get_binding(relation_name).network.private_address
+    elif type == NetworkType.PUBLIC:
+        address = charm.model.get_binding(relation_name).network.ingress_address
```

Comment:
This is not quite accurate - from the docs: 
> Due to things like public/private addresses, NAT and tunneling, the address the charm will bind() to is not always the address other people can use to connect() to the charm. This is just the first address from :attr:`.ingress_addresses`

In any way - we should avoid mixing concerns here. We have 2 separate helpers:
- `get_host_ip`
- `get_host_public_ip` 

Since it's not used elsewhere - We should probably take this out of the current method and keep it simple.